apps/src/aichat/aichatCompletionApi.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  Role,
+  ChatCompletionMessage,
+  AiCustomizations,
+  ChatContext,
+} from './types';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import Lab2Registry from '@cdo/apps/lab2/Lab2Registry';
+
+const CHAT_COMPLETION_URL = '/aichat/chat_completion';
+
+/**
+ * This function sends a POST request to the aichat completion backend controller.
+ */
+
+export async function postAichatCompletionMessage(
+  messagesToSend: AichatCompletionMessage[],
+  temperature: number,
+  chatContext: ChatContext
+): Promise<AichatCompletionMessage | null> {
+  const payload = {
+    inputs: messagesToSend,
+    temperature,
+    chatContext,
+  };
+  const response = await HttpClient.post(
+    CHAT_COMPLETION_URL,
+    JSON.stringify(payload),
+    true,
+    {
+      'Content-Type': 'application/json; charset=UTF-8',
+    }
+  );
+  // For now, response will be null if there was an error.
+  if (response.ok) {
+    return await response.json();
+  } else {
+    return null;
+  }
+}
+
+/**
+ * This function formats chat completion messages including the system prompt, passes them
+ * to `postAichatCompletion`, then returns the status of the response and assistant message
+ * if successful.
+ * TODO: Awaiting details on how to format input for endpoint.
+ */
+export async function getAichatCompletionMessage(
+  aiCustomizations: AiCustomizations,
+  storedMessages: ChatCompletionMessage[],
+  chatContext: ChatContext
+) {
+  const {systemPrompt, temperature, retrievalContexts} = aiCustomizations;
+  const messagesToSend = [
```

Comment:
The goal is to handle whatever reshaping we need to do in Rails. That being said @fisher-alice I think it would be good to send the system prompt and retrieval separately from the chat messages (I think it can be as simple as just sending the whole of `aiCustomizations` as-is). You're right in that system prompt and retrieval will actually just be prepended to the list of messages, but there will probably be more formatting required (like wrapping the prompt in `[INST]` tags) so it would be helpful to have them separated out

Reply:
Gotcha - I wasn't sure if we wanted to format in the front end or back end but will send `aiCustomizations` as is to the backend.

Reply:
Update: sending `aiCustomizations` without `modeCardInfo` as `aiParameters`. Thanks!